internal/config/authorization.go

```diff
@@ -45,64 +55,201 @@ func (c *AuthorizationConfig) setDefaults(v *viper.Viper) error {
+		return errors.New("path must be non-empty string")
 	}
 
-	if a.Local == nil || a.Local.Path == "" {
-		return errors.New("local: path must be non-empty string")
+	if a.PollInterval <= 0 {
+		return errors.New("poll_interval must be greater than zero")
 	}
 
-	if a.PollInterval <= 0 {
-		return errors.New("local: poll_interval must be non-zero")
+	return nil
+}
+
+type AuthorizationSourceBundleConfig struct {
+	Configuration string `json:"configuration,omitempty" mapstructure:"configuration" yaml:"configuration,omitempty"`
+}
+
+func (a *AuthorizationSourceBundleConfig) validate() error {
+	if a == nil {
+		return nil
+	}
+
+	if a.Configuration == "" {
+		return errors.New("configuration must be non-empty string")
 	}
 
 	return nil
 }
 
-// AuthorizationBackend configures the data source backend for
-// additional data supplied to the policy evaluation engine
-type AuthorizationBackend string
+// String returns the configuration as a string in the format expected by the OPA engine
+func (a *AuthorizationSourceBundleConfig) String() string {
+	return a.Configuration
+}
 
-const (
-	AuthorizationBackendLocal = "local"
-)
+// AuthorizationSourceObjectConfig contains authz bundle configuration from object storage.
+type AuthorizationSourceObjectConfig struct {
+	Type ObjectAuthorizationBackendType `json:"type,omitempty" mapstructure:"type" yaml:"type,omitempty"`
+	S3   *S3AuthorizationSource         `json:"s3,omitempty" mapstructure:"s3,omitempty" yaml:"s3,omitempty"`
+	// AZBlob *AZBlobAuthorizationSource        `json:"azblob,omitempty" mapstructure:"azblob,omitempty" yaml:"azblob,omitempty"`
+	// GS     *GSAuthorizationSource            `json:"googlecloud,omitempty" mapstructure:"googlecloud,omitempty" yaml:"googlecloud,omitempty"`
+}
 
-// AuthorizationLocalConfig configures the local backend source
-// for the authorization evaluation engines policies and data
-type AuthorizationLocalConfig struct {
-	Path string `json:"path,omitempty" mapstructure:"path" yaml:"path,omitempty"`
+// validate is only called if authorization.backend == "object"
+func (a *AuthorizationSourceObjectConfig) validate() error {
+	if a == nil {
+		return nil
+	}
+
+	switch a.Type {
+	case S3ObjectAuthorizationBackendType:
+		if a.S3 == nil || a.S3.Bucket == "" {
+			return errors.New("s3 bucket must be specified")
+		}
+
+	default:
+		return errors.New("object storage type must be specified")
+	}
+	return nil
 }
+
+// String returns the configuration as a string in the format expected by the OPA engine
+func (a *AuthorizationSourceObjectConfig) String() string {
+	var (
+		url  strings.Builder
+		tmpl string
+	)
+
+	switch a.Type { //nolint
+	case S3ObjectAuthorizationBackendType:
+		if a.S3.Endpoint != "" {
+			url.WriteString(a.S3.Endpoint)
+		} else {
+			url.WriteString("https://s3")
+			if a.S3.Region != "" {
+				url.WriteString(".")
+				url.WriteString(a.S3.Region)
+			}
+			url.WriteString(".amazonaws.com")
+		}
+
+		tmpl = fmt.Sprintf(`
+services:
```

Comment:
this is gross i'll admit, but its how the OPA SDK has to be configured https://github.com/open-policy-agent/opa/blob/f05497530d337dfd30dbd31851209d3a25c1cf95/sdk/options.go#L27

Reply:
I went looking to see if they at-least had a struct they use to parse it back out, so we could use that and marshal it before we pass it, but even theres is just a pile of json RawMessage:
https://github.com/open-policy-agent/opa/blob/f05497530d337dfd30dbd31851209d3a25c1cf95/config/config.go#L23-L48